modules/programs/rbw.nix

```diff
@@ -0,0 +1,112 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.rbw;
+
+  jsonFormat = pkgs.formats.json { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+  settingsModule = with lib;
+    types.submodule {
+      freeformType = jsonFormat.type;
+      options = {
+        email = mkOption {
+          type = types.str;
+          example = "name@example.com";
+          description = "The email address for your bitwarden account.";
+        };
+
+        base_url = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          example = "bitwarden.example.com";
+          description =
+            "The base-url for a self-hosted bitwarden installation.";
+        };
+
+        identity_url = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          example = "identity.example.com";
+          description = "The identity url for your bitwarden installation.";
+        };
+
+        lock_timeout = mkOption {
+          type = types.ints.unsigned;
+          default = 3600;
+          example = 300;
+          description = ''
+            The amount of time that your login information should be cached.
+          '';
+        };
+
+        pinentry = mkOption {
+          type = types.enum pkgs.pinentry.flavors;
```

Comment:
Unfortunately, this would make it hard to use with `pinentry_mac`
in darwin we usually use `${pkgs.pinentry_mac}/Applications/pinentry-mac.app/Contents/MacOS/pinentry-mac`